ffi/src/winscard/system_scard/card.rs

```diff
@@ -164,34 +170,34 @@ impl WinScard for SystemScard {
             )?;
         }
 
-        // SAFETY: A slice creation is safe in this context because the `reader_name` pointer is
-        // a local pointer that was initialized by the `SCardStatus` function.
-        let multi_string_buffer = unsafe { from_raw_parts(reader_name, reader_name_len.try_into()?) };
+        let multi_string_buffer = &reader_name[0..reader_name_len.try_into()?];
+
         let readers = if let Ok(readers) = parse_multi_string_owned(multi_string_buffer) {
             readers
         } else {
-            try_execute!(
-                // SAFETY: This function is safe to call because `self.h_card_context` is always a valid handle.
-                unsafe { (self.api.SCardFreeMemory)(self.h_card_context, reader_name as *const _) },
-                "SCardFreeMemory failed"
-            )?;
-
             return Err(Error::new(
                 ErrorKind::InternalError,
                 "returned reader is not valid UTF-8",
             ));
         };
 
-        try_execute!(
-            // SAFETY: This function is safe to call because `self.h_card_context` is always a valid handle.
-            unsafe { (self.api.SCardFreeMemory)(self.h_card_context, reader_name as *const _) },
-            "SCardFreeMemory failed"
-        )?;
+        atr.truncate(atr_len.try_into()?);
 
         let status = Status {
             readers,
+            #[cfg(not(target_os = "windows"))]
+            state: {
+                use crate::winscard::pcsc_lite::State;
+
+                State::from_bits(state).unwrap_or(State::Specific).into()
+            },
```

Comment:
style: In this case, I suggest this more compact form:
```suggestion
            state: crate::winscard::pcsc_lite::State::from_bits(state).unwrap_or(State::Specific).into(),
```

Reply:
done